analysis/src/CompletionBackEnd.ml

```diff
@@ -992,6 +992,18 @@ let rec extractRecordType ~env ~package (t : Types.type_expr) =
     | _ -> None)
   | _ -> None
 
+let rec extractVariantType ~env ~package (t : Types.type_expr) =
```

Comment:
There's probably something like that already. Or maybe not.
Anyhow no need to worry about this for now.

Reply:
I refactored this into a general one for extracting a type, so the code is slightly more ready for completing more things than just variants.